civis/tests/test_utils.py

```diff
@@ -260,3 +288,75 @@ def test_no_retry_on_connection_error(mock_session):
             pass
 
         assert session_context.send.call_count == expected_call_count
+
+
+@mock.patch('civis._utils.open_session')
+def test_retry_respect_retry_after_headers(mock_session):
+    expected_call_count = 0
+    max_calls = 3
+    retry_after = 3
+    api_response = {'key': 'value'}
+    session_context = mock_session.return_value.__enter__.return_value
+    session_context.send.return_value.json.return_value = api_response
+
+    session_context.send.return_value.status_code = 429
+    session_context.send.return_value.headers = {
+        'Retry-After': str(retry_after)
+    }
+
+    for verb in ['HEAD', 'TRACE', 'GET', 'PUT', 'OPTIONS', 'DELETE', 'POST']:
+        expected_call_count += max_calls
+
+        request_info = dict(
+            params={'secondParameter': 'b', 'firstParameter': 'a'},
+            json={},
+            url='https://api.civisanalytics.com/wobble/wubble',
+            method=verb
+        )
+
+        request = Request(**request_info)
+        pre_request = session_context.prepare_request(request)
+
+        start_time = datetime.now().timestamp()
+        retry_request(verb, pre_request, session_context, max_calls)
+        end_time = datetime.now().timestamp()
+        duration = end_time - start_time
+
+        assert session_context.send.call_count == expected_call_count
+        assert floor(duration) == retry_after * (max_calls - 1)
+
+
+@mock.patch('civis._utils.open_session')
+def test_retry_respect_retry_after_headers_lowercase_verb(mock_session):
+    expected_call_count = 0
+    max_calls = 3
+    retry_after = 3
+    api_response = {'key': 'value'}
+    session_context = mock_session.return_value.__enter__.return_value
+    session_context.send.return_value.json.return_value = api_response
+
+    session_context.send.return_value.status_code = 429
+    session_context.send.return_value.headers = {
+        'Retry-After': str(retry_after)
+    }
+
+    for verb in ['head', 'trace', 'get', 'put', 'options', 'delete', 'post']:
```

Comment:
This function is the same as the one above except for the name and this line, right? Could you just put the lowercase verb strings in the list in the other function ([here](https://github.com/civisanalytics/civis-python/pull/413/files#diff-f2df2ec5c411a1d0c536fcfb23aa66d7196fa463abd9d22460617bb37d667db2R307))?  `pytest.mark.parametrize` would be another option, but that seems like overkill.